### PR TITLE
Add tests for InfiniBand testing

### PR DESCRIPTION
**Pull request by frankenmichl**
Add a new load_baremetal_tests to products/sle/main.pm, which for now
only loads the InfiniBand testsuite (and sets up the barriers needed for
this).
Add tests/kernel/ib_tests.pm that allows to execute a testsuite for
InfiniBand testing. The testsuite is currently under development at
https://gitlab.suse.de/NMoreyChaisemartin/hpc-testing/.

- Related ticket: https://progress.opensuse.org/issues/25678
- Needles: -
- Verification run: http://loewe.arch.suse.de/tests/749 - note some testcases are failing

Signed-off-by: Michael Moese <mmoese@suse.de>

### CODE REVIEW

File: tests/kernel/ib_tests.pm [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+    power_action('poweroff', observe => 0, keepconsole => 0);
+}
+
+sub test_flags {
+    return {fatal => 1};
+}
+
+sub post_fail_hook {
+    script_run('tr -cd \'\11\12\15\40-\176\' < results/TEST-ib-test.xml > /tmp/results.xml');
```
Comment (nicksinger) [inline tests/kernel/ib_tests.pm]: A short explanation of this magic set of characters and where they come from would be nice

File: tests/kernel/ib_tests.pm [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+The workers with the InfiniBand HCA's need a special worker class, in this case
+we assume it is "64bit-ipmi_infiniband". The third worker just needs a different 
+worker class, something like "64bit-ipmi" should be fine.
+
+=head2 openQA test suites
+As the test is executed on three hosts, three test suites should be created:
+
+=head3 ibtest-master	
+ADDONURL_SDK=ftp://openqa.suse.de/SLE-15-SDK-POOL-%ARCH%-Build%BUILD%-Media1/
+GA_REPO="http://download.suse.de/ibs/SUSE:/SLE-15:/GA/standard/SUSE:SLE-15:GA.repo"
```
Comment (nicksinger) [inline tests/kernel/ib_tests.pm]: Not sure if everybody is fine with having internal URLs in public repos. I'd repalce it with an example URL but no hard feelings

File: tests/kernel/ib_tests.pm (hunk 1 above)
Comment (frankenmichl) [reply tests/kernel/ib_tests.pm]: Sure, I could add a comment. This line just removes non-printable characters.

File: tests/kernel/ib_tests.pm (hunk 2 above)
Comment (frankenmichl) [reply tests/kernel/ib_tests.pm]: Valid point. I'll change it.

Comment (okurz) [reply tests/kernel/ib_tests.pm]: Or simply follow "Factory first" and provide primarily tests that work for openSUSE Tumbleweed. Then you can easily use public OBS repos as reference anyway :)

File: products/sle/main.pm [hunk 3]
```diff
@@ -677,10 +699,14 @@ if (is_jeos) {
     loadtest "console/suseconnect_scc";
 }
 
+
```
Comment (okurz) [inline products/sle/main.pm]: That's not a related change. If you want to change the structure for a good reason please make it at least its own commit with a good reasoning

File: products/sle/main.pm [hunk 4]
```diff
@@ -1120,4 +1146,5 @@ else {
 
 load_common_opensuse_sle_tests;
 
+
```
Comment (okurz) [inline products/sle/main.pm]: please revert this change

File: tests/kernel/ib_tests.pm [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: run InfiniBand test suite hpc-testing
+#          https://gitlab.suse.de/NMoreyChaisemartin/hpc-testing
```
Comment (okurz) [inline tests/kernel/ib_tests.pm]: better also here not reference any non-public repos

File: tests/kernel/ib_tests.pm [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+# Maintainer: Michael Moese <mmoese@suse.de>
+#             Nick Singer <nsinger@suse.de>
```
Comment (okurz) [inline tests/kernel/ib_tests.pm]: Please use a comma-separated list which is the common format so far

File: tests/kernel/ib_tests.pm [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    }
+    elsif ($role eq 'IBTEST_SLAVE1' || $role eq 'IBTEST_SLAVE2') {
+        ibtest_slave;
+    }
+
+    power_action('poweroff', observe => 0, keepconsole => 0);
+}
+
+sub test_flags {
+    return {fatal => 1};
```
Comment (okurz) [inline tests/kernel/ib_tests.pm]: why does this test module need to be fatal?

**Comment by nicksinger** [comment]
@okurz unfortunately we've to focus on SLE-only here since the hardware is and will never become available in the openSUSE network. We're aware that this is not nice but right now it is the only possible solution.
Do you know if there is any shared hardware between OSD and O3? If so, it could work out. If this would be the first case, I doubt that it's even possible (not because of technical reasons but because of the strong separation between these two networks)

File: tests/kernel/ib_tests.pm (hunk 2 above)
Comment (frankenmichl) [reply tests/kernel/ib_tests.pm]: Unfortunately, factory first will not work here, as the machines required for this testsuite are not accessible from openqa.opensuse.org

**Comment by okurz** [comment]
Everything looks fine for me

File: products/sle/main.pm [hunk 8]
```diff
@@ -646,6 +646,28 @@ sub prepare_target {
     }
 }
 
+sub is_baremetal_test {
+    return get_var('IBTESTS');
```
Comment (jlausuch) [inline products/sle/main.pm]: Maybe we could be more generic and create a global variable for BM. 
NFV tests are also for BM but can't use this function. What about reading if variable BAREMETAL=1 is set and return true if so (or something similar).

File: products/sle/main.pm [hunk 9]
```diff
@@ -646,6 +646,28 @@ sub prepare_target {
...
+}
+
+sub load_baremetal_tests {
```
Comment (jlausuch) [inline products/sle/main.pm]: +1 to this function. I will adapt NFV tests to use this function so we don't repeat always load_boot_tests(), load_inst_tests() and load_reboot_tests().

File: products/sle/main.pm (hunk 8 above)
Comment (frankenmichl) [reply products/sle/main.pm]: I thought about this. However, setting up these tests already involves correctly setting quite some Variables. This is why I decided to do it like it is already done for the kernel tests. I'm not sure if another variable would be helpful.
